AliceAdventure/app/Editor/Pages/css/layout.css

```diff
@@ -86,31 +84,29 @@ li a {
 }
 
 canvas {
-  /* width: 80% !important; */
-  /* height: auto;
-  margin: 0 auto; */
+  /* These attributes are fixing the canvas ratio to 16 : 9. 
+  In the original file, the size is 1024 * 576. 
+  The viewport is 1904 * 1021 (window size is 1920 * 1080).
+  So to make sure the canvas has the same ratio in different window size, we need a absolute length which I use vw (viewport width) here.
+  The width should be 53.78vw because 0.5378 * 1904 = 1024.
+  And the ratio is 16 : 9, so height should be 30.25125vw.
+  However, since we do not consider the height of the viewport, the canvas will overflow in some case.
+  So we need restrictions, this time we need to make bounds depending on vh.
+  If you see the following code in .grid-container, then we can calculate the ratio of canvas height depending on origin grid layout design which is 58.77vh.
+  So here I choose to make 55vh as a max height and calculate the another one depening on the ratio.
+  In this case, the canvas will not overflow and maintain a ratio of 16 : 9. */
```

Comment:
That's awesome thanks! I'm curious, on macOS, is the viewport still 1904 * 1021? (I suppose the menu bar will be thinner)

Reply:
You are right, on the masOS the application shows the menu bar on the top, it's 1920 * 1058. I will talk to the lead to see if I need to make another pull request to fix it.